gems/sorbet-runtime/lib/types/configuration.rb

```diff
@@ -290,6 +290,51 @@ def self.log_info_handler(str, extra)
     end
   end
 
+  # Set a handler for deserialization errors
+  #
+  # These generally shouldn't stop execution of the program, but rather inform
+  # some party of the error to action on later.
+  #
+  # @param [Lambda, Proc, Object, nil] value Proc that handles the error
+  #   report (pass nil to reset to default behavior)
+  #
+  # Parameters passed to value.call:
+  #
+  # @param [Class] klass The class of the top-level object being deserialized
+  # @param [String] prop_name The name of the prop being deserialized
+  # @param [Object] value The prop value being deserialized
+  # @param [Error] orig_error The original error raised during deserialization
+  #
+  # @example
+  #   T::Configuration.deserialization_error_handler = lambda do |klass, prop_name, value, orig_error}
+  #     puts "Error deserializing #{prop_name} on #{klass}: #{orig_error.message}
+  #   end
+  def self.deserialization_error_handler=(value)
+    validate_lambda_given!(value)
+    @deserialization_error_handler = value
+  end
+
+  private_class_method def self.deserialization_error_handler_default(klass, prop_name, value, orig_error)
+    soft_assert_handler(
+      'Deserialization error (probably unexpected stored type)',
+      storytime: {
+        klass: klass,
+        prop: prop_name,
+        value: value,
+        error: orig_error.message,
+        notify: 'djudd'
```

Comment:
@djudd-stripe What is the deprecation plan for this soft assert? is it still firing? When do we plan to either handle the fact this soft assert fires, or convert the soft assert to a hard assert?

Reply:
It looks like this might be almost eradicated?

Splunk search: http://go/arsoDxmZ

So I wonder if we can:

- factor this logic to a private helper function, and
- convert it to a hard error

And then not have to worry about this being a public API, nor having the `notify: 'djudd'` in the codebase anymore.

Reply:
That's a lot less than last time I checked, yay!

I have no plans to follow up on this. I was going to originally but the notification wasn't actually working - it gets overridden to notify: false in T::Configuration in pay-server IIRC - so I didn't realize it was triggering until the time I had to work on it was used up.

But your plan makes sense.